taxonomy.php

```diff
@@ -16,20 +16,40 @@
 
 $context              = Timber::get_context();
 $context['page_type'] = get_queried_object();
-$pagetype_posts       = Timber::get_posts();
 
-if ( $pagetype_posts ) {
-	foreach ( $pagetype_posts as $pagetype_post ) {
-
-		$author_override                = get_post_meta( $pagetype_post->ID, 'p4_author_override', true );
-		$pagetype_post->author          = '' === $author_override ? get_the_author_meta( 'display_name', $pagetype_post->post_author ) : $author_override;
-		$pagetype_post->author_url      = '' === $author_override ? get_author_posts_url( $pagetype_post->post_author ) : '#';
-		$pagetype_post->author_override = $author_override;
-	}
+function setAuthorOverride($post) {
+	$author_override       = get_post_meta( $post->ID, 'p4_author_override', true );
+	$post->author          = '' === $author_override ? get_the_author_meta( 'display_name', $post->post_author ) : $author_override;
+	$post->author_url      = '' === $author_override ? get_author_posts_url( $post->post_author ) : '#';
+	$post->author_override = $author_override;
+	return $post;
```

Comment:
Since we are using author_override more and more often I suggest splitting this function and putting it's parts in P4_Post.

`P4_Post `is subclass of `Timber\Post`
Timber\Post has an author object already and custom array property which contains custom post metadata.
[author](https://timber.github.io/docs/reference/timber-post/#author)

So I'm thinking about adding 2 properties in P4_Post
author_override
author_override_url
and then setting the first with
`$this->author_override = $this->custom['p4_author_override'] `(if it exists)